scipy/stats/_qmc.py

```diff
@@ -883,6 +977,177 @@ def reset(self):
         return self
 
 
+class OptimalDesign(QMCEngine):
+    """Optimal design.
+
+    Optimize the design by doing random permutations to lower the centered
+    discrepancy [1]_.
+
+    The specified optimization `method` is used to select a new set of
+    permutations to perform. If `method` is None, *basinhopping* optimization
+    is used. `niter` set of permutations are performed.
+
+    Centered discrepancy-based design shows better space filling robustness
+    toward 2D and 3D subprojections. Distance-based design shows better space
+    filling but less robustness to subprojections [2]_.
+
+    Parameters
+    ----------
+    d : int
+        Dimension of the parameter space.
+    start_design : array_like (n, d), optional
+        Initial design of experiment to optimize. `LatinHypercube`
+        is used to generate a first design otherwise.
+    niter : int, optional
+        Number of iterations to perform. Default is 1.
+    method : callable ``f(func, x0, bounds)``, optional
+        Optimization function used to search new samples. Default to
+        `scipy.optimize.basinhopping` optimization.
+    seed : {None, int, `numpy.random.Generator`}, optional
+        If `seed` is None the `numpy.random.Generator` singleton is used.
+        If `seed` is an int, a new ``Generator`` instance is used,
+        seeded with `seed`.
+        If `seed` is already a ``Generator`` instance then that instance is
+        used.
+
+    References
+    ----------
+    .. [1] Fang et al. Design and modeling for computer experiments,
+       Computer Science and Data Analysis Series, 2006.
+    .. [2] Damblin et al., "Numerical studies of space filling designs:
+       optimization of Latin Hypercube Samples and subprojection properties",
+       Journal of Simulation, 2013.
+
+    Examples
+    --------
+    Generate samples from an optimal design.
+
+    >>> from scipy.stats import qmc
+    >>> sampler = qmc.OptimalDesign(d=2, seed=12345)
+    >>> sample = sampler.random(n=5)
+    >>> sample
+    array([[0.84052691, 0.53664833],
+           [0.1545328 , 0.36265316],
+           [0.52177809, 0.93343721],
+           [0.68033825, 0.06474907],
+           [0.26544879, 0.61163943]])
+
+    Compute the quality of the sample using the discrepancy criterion.
+
+    >>> qmc.discrepancy(sample)
+    0.01602810655510689
+
+    You can possibly improve the quality of the sample by performing more
+    optimization iterations by using `niter`:
+
+    >>> sampler_2 = qmc.OptimalDesign(d=2, niter=5, seed=12345)
+    >>> sample_2 = sampler_2.random(n=5)
+    >>> qmc.discrepancy(sample_2)
+    0.01600398742881648
+
+    """
+
+    def __init__(self, d, start_design=None, niter=1, method=None, seed=None):
+        super().__init__(d=d, seed=seed)
+        self.start_design = start_design
```

Comment:
Ensure that it's an array, and we want to copy it. Go ahead and save it into `self.best_doe` initially rather than having two attributes, `best_doe` and `start_design`.